sql/src/main/java/io/crate/analyze/where/WhereClauseNullReplacer.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.analyze.where;
+
+import io.crate.analyze.symbol.Function;
+import io.crate.analyze.symbol.FunctionCopyVisitor;
+import io.crate.analyze.symbol.Literal;
+import io.crate.analyze.symbol.Symbol;
+import io.crate.execution.expression.operator.Operators;
+import io.crate.execution.expression.predicate.NotPredicate;
+import io.crate.metadata.FunctionInfo;
+
+/**
+ * Inside a where clause NULL values as logical operator arguments can be treated like boolean FALSE.
```

Comment:
This would also apply to a HAVING clause - as such I think we should try to avoid the "WhereClause" naming here. E.g. simply call this `NullEliminator` or something like that.